pkgs/development/compilers/bambu/default.nix

```diff
@@ -0,0 +1,257 @@
+      for f in $out/bin/*; do
+        if [ -f "$f" ]; then
+          wrapProgram "$f" \
+            --set AR_FOR_TARGET ar \
+            --set AS_FOR_TARGET as \
+            --set CC_FOR_TARGET gcc \
+            --set CXX_FOR_TARGET g++ \
+            --set LD_FOR_TARGET ld \
+            --set NIX_BINTOOLS_FOR_TARGET ${i686gcc.bintools} \
+            --set NIX_BINTOOLS_WRAPPER_TARGET_TARGET_i686_unknown_linux_gnu 1 \
+            --set NIX_CC_FOR_TARGET ${i686gcc} \
+            --set NIX_CC_WRAPPER_TARGET_TARGET_i686_unknown_linux_gnu 1 \
+            --set NM_FOR_TARGET nm \
+            --set OBJCOPY_FOR_TARGET objcopy \
+            --set OBJDUMP_FOR_TARGET objdump \
+            --set RANLIB_FOR_TARGET ranlib \
+            --set READELF_FOR_TARGET readelf \
+            --set SIZE_FOR_TARGET size \
+            --set STRINGS_FOR_TARGET strings \
+            --set STRIP_FOR_TARGET strip \
+            --prefix HOST_PATH ":" ${i686gcc.bintools}/bin \
+            --prefix HOST_PATH ":" ${i686gcc}/bin \
+            --prefix PATH ":" ${i686gcc.bintools}/bin \
+            --prefix PATH ":" ${i686gcc}/bin \
+            --prefix NIX_LDFLAGS_FOR_TARGET " " "-L${llvmPackages.clang}/lib -L${llvmPackages.clang.cc.libllvm}/lib -L${pkgsi686Linux.buddy}/lib -L${libmpc}/lib -L${gmp}/lib -L${mpfr}/lib -L${boost}/lib -L${libxml2}/lib -L${xz}/lib -L${zlib}/lib -L${icu}/lib -L${suitesparse}/lib -L${mpfi}/lib -L${glpk}/lib  -L${i686gcc.cc}/lib" \
+            --prefix NIX_CFLAGS_COMPILE_FOR_TARGET " " "-I${llvmPackages.clang.cc.dev}/include -I${llvmPackages.clang.cc.libllvm.dev}/include -I${pkgsi686Linux.buddy}/include -I${libmpc}/include -I${gmp.dev}/include -I${mpfr.dev}/include -I${boost.dev}/include -I${libxml2.dev}/include -I${xz.dev}/include -I${zlib.dev}/include -I${icu.dev}/include -I${suitesparse.dev}/include -I${mpfi}/include -I${glpk}/include -I${i686gcc.cc}/include"
+        fi
+      done
+    '';
+  };
+
+  clangFrontendsFlags = builtins.map (llvmPackages: ''--with-clang${lib.versions.major llvmPackages.clang.version}=${combineClangAndLLVM llvmPackages}/bin/clang '') clangFrontends;
+  gccFrontendsFlags = builtins.map (gcc: ''--with-gcc${lib.versions.major gcc.version}=${wrapGcc gcc}/bin/gcc '') gccFrontends;
+in
+# Assert that there is at least one clang and GCC frontend specified
+assert lib.assertMsg (builtins.length clangFrontendsFlags != 0) "There needs to be at least one clang frontend specified";
+assert lib.assertMsg (builtins.length gccFrontendsFlags != 0) "There needs to be at least one GCC frontend specified";
+# Assert that the passed clang and GCC frontends are i686
+assert builtins.all
+  (llvmPackages:
+  lib.assertMsg (llvmPackages.clang.system == "i686-linux") "Bambu frontends have to be i686 but ${llvmPackages.clang.name} is ${llvmPackages.clang.system}")
+  clangFrontends;
+assert builtins.all
+  (gcc:
+  lib.assertMsg (gcc.system == "i686-linux") "Bambu frontends have to be i686 but ${gcc.name} is ${gcc.system}")
```

Comment:
`.system` is only the system on which the derivation will be built on, not the host.

Use `.stdenv.hostPlatform.system` instead